pkg/util/cloudproviders/cloudfoundry/cccache.go

```diff
@@ -180,7 +180,76 @@ func (ccc *CCCache) UpdatedOnce() <-chan struct{} {
 	return ccc.updatedOnce
 }
 
-func (ccc *CCCache) waitForResource(guid string) {
+// getResource looks up the given resourceName/GUID in the CCCache
+// If not found and refreshOnCacheMiss is enabled, it will use the fetchFn function to fetch the resource from the CAPI
+func getResource[T any](ccc *CCCache, resourceName, guid string, cache map[string]T, fetchFn func(string) (T, error)) (T, error) {
+	ccc.RLock()
+	resource, ok := cache[guid]
+	ccc.RUnlock()
+
+	if !ok {
+		if !ccc.refreshCacheOnMiss {
+			return resource, fmt.Errorf("refreshCacheOnMiss is disabled, could not find resource '%s' with guid '%s' in cloud controller cache", resourceName, guid)
```

Comment:
```suggestion
			return resource, fmt.Errorf("Could not find resource '%s' with guid '%s' in cloud controller cache. Consider enabling `refreshCacheOnMiss`", resourceName, guid)
```
What do you think about this wording?